src/sdg/producten/models/managers.py

```diff
@@ -1,4 +1,131 @@
+from datetime import date
+
 from django.db import models
+from django.db.models import Case, F, Prefetch, Subquery, When
+
+
+class ProductQuerySet(models.QuerySet):
+    def active(self, active_on=None):
+        """
+        An active product is a product with its product version and its
+        translations as it was most recently published.
+
+        Unpublished (future versions or concept versions) products are not
+        included.
+
+        This function optimizes the queryset to prefetch the `ProductVersie`
+        that is active on given `active_on` date, including the appropriate
+        prefetched translations. The prefetched active version is available
+        via `Product.active_version` and is always a list, containing 0 or 1
+        `ProductVersie`.
+        """
+        from sdg.producten.models import ProductVersie
+
+        if active_on is None:
+            active_on = date.today()
+
+        subquery = Subquery(
+            ProductVersie.objects.exclude(publicatie_datum=None)
+            .filter(publicatie_datum__lte=active_on)
+            .order_by("-versie")
+            .values_list("pk", flat=True)[:1]
+        )
+
+        return self.prefetch_related(
+            Prefetch(
+                "versies",
+                to_attr="active_versions",
+                queryset=ProductVersie.objects.filter(pk__in=subquery).prefetch_related(
+                    "vertalingen"
+                ),
+            )
+        )
+
+    def most_recent(self):
+        """
+        The most recent product is the product that was last published
+        (including future publications) or a concept.
+
+        This function optimizes the queryset to prefetch the most recent
+        `ProductVersie`, including the appropriate prefetched translations. The
+        prefetched most recent version is available via
+        `Product._most_recent_version` and is always a list, containing 0 or 1
+        `ProductVersie`.
+        """
+        from sdg.producten.models import ProductVersie
+
+        subquery = Subquery(
+            ProductVersie.objects.order_by("-versie").values_list("pk", flat=True)[:1]
+        )
+
+        return self.prefetch_related(
+            Prefetch(
+                "versies",
+                to_attr="_most_recent_version",
+                queryset=ProductVersie.objects.filter(pk__in=subquery).prefetch_related(
+                    "vertalingen"
+                ),
+            )
+        )
+
+    def annotate_is_reference(self):
```

Comment:
👍 but is it used?

In any case, please use a similar construction as done here: https://github.com/maykinmedia/sdg-invoervoorziening/pull/253/files#diff-afbcf6c0a991e8802d5b7eb70227b5cb12d402dac45ef47ad635b5547271a9d7R213

Add a property `is_reference`, annotate with `_is_reference` (for consistancy) so you are sure you can always use `Product.is_reference` and it will just work (annotated or not).